src/v/datalake/batching_parquet_writer.cc

```diff
@@ -210,9 +210,9 @@ batching_parquet_writer_factory::batching_parquet_writer_factory(
   , _row_count_threshold{row_count_threshold}
   , _byte_count_treshold{byte_count_threshold} {}
 
-ss::future<result<std::unique_ptr<data_writer>, data_writer_error>>
+ss::future<result<ss::shared_ptr<data_writer>, data_writer_error>>
 batching_parquet_writer_factory::create_writer(iceberg::struct_type schema) {
-    auto ret = std::make_unique<batching_parquet_writer>(
+    auto ret = ss::make_shared<batching_parquet_writer>(
```

Comment:
why do we need a shared pointer to be returned from the factory ? Do we require this shared ownership semantics ?

Reply:
Yes, the caller of get_writer needs to hold a reference to this, and we need to keep it in this data structure.

Previously it was a unique_ptr and get_writer returned a reference, but a reference cannot be wrapped in a result.